### PR TITLE
Add unit tests for ModelExpressionPass

### PR DESCRIPTION
**Pull request by rynowak**
/cc @NTaylorMullen @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs [hunk 1]
```diff
@@ -55,10 +58,35 @@ public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
                         });
                     }
                     else
-                    {
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs]: Unit testing found a bug here. We were just copying all of the nodes into the new expression, but that isn't quite right. We also uncovered a bug in Razor. Once this is addressed, won't need to handle "HTML" here (note the scare quotes).

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs [hunk 2]
```diff
@@ -55,10 +58,35 @@ public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
...
+                    {
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs]: Format

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                "[Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]" + Environment.NewLine +
+                "public PropertyType<dynamic> PropertyName { get; private set; }",
+                statement.Content);
+        }
+
+        [Fact]
+        public void InjectDirectivePass_Execute_ExpandsTModel_WithModelType()
+        {
+            // Arrange
+            var codeDocument = CreateDocument(@"
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs]: Could we have the inverse order too:

```
@model ModelType
@inject PropertyType<TModel> PropertyName
```

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            pass.Execute(codeDocument, irDocument);
+
+            // Assert
+            var @class = FindClassNode(irDocument);
+            Assert.NotNull(@class);
+            Assert.Equal(2, @class.Children.Count);
+
+            var statement = Assert.IsType<CSharpStatementIRNode>(@class.Children[1]);
+            Assert.Equal(
+                "[Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]" + Environment.NewLine +
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs]: Do we do anything funky with line endings anymore? Is it NewLine or `\r\n` all the time?

Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs]: This won't affect any designtime stuff, it's in generated code that no one will see.

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs (hunk 3 above)
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs]: Sure

**Comment by rynowak** [comment]
705369ee51c73c9b5179728c0e7bd934b19b1552

**Pull request closed without merging**